src/plugins/charts/public/static/components/color_picker.tsx

```diff
@@ -74,54 +74,87 @@ export const legendColors: string[] = [
+   */
+  useLegacyColors?: boolean;
+  /**
+   * Defines if the default color is overwritten. Defaults to true.
+   */
+  colorIsOverwritten?: boolean;
 }
 
-export const ColorPicker = ({ onChange, color: selectedColor, id, label }: ColorPickerProps) => (
-  <div className="visColorPicker">
-    <span id={`${id}ColorPickerDesc`} className="euiScreenReaderOnly">
-      <FormattedMessage
-        id="charts.colorPicker.setColor.screenReaderDescription"
-        defaultMessage="Set color for value {legendDataLabel}"
-        values={{ legendDataLabel: label }}
-      />
-    </span>
-    <div className="visColorPicker__value" role="listbox">
-      {legendColors.map((color) => (
-        <EuiIcon
-          role="option"
-          tabIndex={0}
-          type="dot"
-          size="l"
-          color={selectedColor}
-          key={color}
-          aria-label={color}
-          aria-describedby={`${id}ColorPickerDesc`}
-          aria-selected={color === selectedColor}
-          onClick={(e) => onChange(color, e)}
-          onKeyPress={(e) => onChange(color, e)}
-          className={classNames('visColorPicker__valueDot', {
-            // eslint-disable-next-line @typescript-eslint/naming-convention
-            'visColorPicker__valueDot-isSelected': color === selectedColor,
-          })}
-          style={{ color }}
-          data-test-subj={`visColorPickerColor-${color}`}
+const euiColors = euiPaletteColorBlind({ rotations: 4, order: 'group' });
+
+export const ColorPicker = ({
+  onChange,
+  color: selectedColor,
+  id,
+  label,
+  useLegacyColors = true,
+  colorIsOverwritten = true,
+}: ColorPickerProps) => {
+  const legendColors = useLegacyColors ? legacyColors : euiColors;
+
+  return (
+    <div className="visColorPicker">
+      <span id={`${id}ColorPickerDesc`} className="euiScreenReaderOnly">
+        <FormattedMessage
+          id="charts.colorPicker.setColor.screenReaderDescription"
+          defaultMessage="Set color for value {legendDataLabel}"
+          values={{ legendDataLabel: label }}
         />
-      ))}
+      </span>
+      <div className="visColorPicker__value" role="listbox">
+        {legendColors.map((color) => (
+          <EuiIcon
+            role="option"
+            tabIndex={0}
+            type="dot"
+            size="l"
+            color={selectedColor}
+            key={color}
+            aria-label={color}
+            aria-describedby={`${id}ColorPickerDesc`}
+            aria-selected={color === selectedColor}
+            onClick={(e) => onChange(color, e)}
+            onKeyPress={(e) => onChange(color, e)}
+            className={classNames('visColorPicker__valueDot', {
+              // eslint-disable-next-line @typescript-eslint/naming-convention
+              'visColorPicker__valueDot-isSelected': color === selectedColor,
+            })}
+            style={{ color }}
+            data-test-subj={`visColorPickerColor-${color}`}
+          />
```

Comment:
Similar to above, let's get rid of a lot of this custom work and rely on basic HTML. 

```suggestion
		<label key={color}>
		  <input type="radio" onChange={(e) => onChange(color, e)}/>
          <EuiIcon
            type="dot"
            size="l"
            color={selectedColor}
            className={classNames('visColorPicker__valueDot', {
              // eslint-disable-next-line @typescript-eslint/naming-convention
              'visColorPicker__valueDot-isSelected': color === selectedColor,
            })}
            style={{ color }}
            data-test-subj={`visColorPickerColor-${color}`}
          />
          <EuiScreenReaderOnly><span>{color}</span></EuiScreenReaderOnly>
        </label>
```

Notice how we got rid of every `aria-*` attribute? If that's something that can be achieved, it should be a goal. `aria-*` attributes are a last resort when other options aren't available.

Hopefully, this should also fix the keyboard controls because currently I can't use this component with a keyboard at all (only tested in Firefox).

Reply:
Oh, just a warning, this code hasn't been tested. There might be some issues in it, notably: 
- you'll need to hide the `input` so that there isn't a visible radio button (you might be able to wrap it in `<EuiScreenReaderOnly>` and have that work but I wasn't sure and I didn't want to lead you down the wrong path inadvertently)
- you'll probably also need to override the default focus state of the native radio button